pkg/controller/inferenceservice/controller_test.go

```diff
@@ -374,13 +395,70 @@ func TestInferenceServiceWithOnlyPredictor(t *testing.T) {
 		if len(events.Items) == 0 {
 			return fmt.Errorf("Test %q failed: no events were created", serviceName)
 		}
+		var passes = 0
 		for _, event := range events.Items {
-			if event.Reason == string(kfserving.InferenceServiceNotReadyState) &&
-				event.Type == v1.EventTypeWarning {
-				return nil
+			if event.Reason == string(kfserving.InferenceServiceReadyState) &&
```

Comment:
We can make a function `getEvents` which returns an array of `event.Reason` that is sorted by timestamp, so we can assert with the expected state sequence and make the tests less lengthy.